packages/meditrak-server/src/routes/GETSurveyResponses/assertSurveyResponsePermissions.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import keyBy from 'lodash.keyby';
+
+export const hasSurveyResponsePermissions = async (accessPolicy, models, surveyResponse) => {
+  const entity = await models.entity.findById(surveyResponse.entity_id);
+  const survey = await models.survey.findById(surveyResponse.survey_id);
+  const permissionGroup = await models.permissionGroup.findById(survey.permission_group_id);
+
+  return accessPolicy.allows(entity.country_code, permissionGroup.name);
+};
+
+export const assertSurveyResponsePermissions = async (accessPolicy, models, surveyResponse) => {
+  const hasPermission = await hasSurveyResponsePermissions(accessPolicy, models, surveyResponse);
+  if (!hasPermission) {
+    throw new Error('You do not have permissions for this survey response');
+  }
+  return true;
+};
+
+export const filterSurveyResponsesByPermissions = async (accessPolicy, surveyResponses, models) => {
```

Comment:
I've also just realized something. If we have BES Admin (Super user) access we should be able to see all the survey responses without any restrictions. We can add the below code before checking anything:

```
if (accessPolicy.allowsSome(null, BES_ADMIN_PERMISSION_GROUP)) {
	return surveyResponses;
}
```
Sorry for not thinking about that earlier :(

Reply:
Added